crates/revm/src/evm_impl.rs

```diff
@@ -229,6 +236,9 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> Transact<DB::Error>
                 journal,
             )?;
 
+            let Some(tx_l1_cost) = tx_l1_cost else {
+                panic!("[OPTIMISM] L1 Block Info could not be loaded from the DB.")
+            };
```

Comment:
I think we'd want to move this to the top - we should panic early right?

Reply:
Then we can also just concretely make `l1_block_info` of type `L1BlockInfo` rather than the `Option<L1BlockInfo>` and avoid needing to panic in the `finalize` function, and changing its signature to:
```rust
    fn finalize<SPEC: Spec>(
        &mut self,
        gas: &Gas,
        #[cfg(feature = "optimism")] l1_block_info: &optimism::L1BlockInfo,
    ) -> (HashMap<B160, Account>, Vec<Log>, u64, u64) {
```

Reply:
Ah yeah - primary reason this is here is because it's under the runtime `optimism` option. We can throw it in the top, but would add an extra branch. Failing early shouldn't be a huge problem, not too far from where it's instantiated

Reply:
It does reduce the code size though since you only now have to panic once rather than again in the `finalize` function

Reply:
Ah I see what you mean, I think it's reasonable then to do this